src/python/zquantum/core/wip/estimators/estimation.py

```diff
@@ -0,0 +1,236 @@
+from typing import List, Tuple, Optional
+from openfermion import QubitOperator, IsingOperator
+import pyquil
+import numpy as np
+
+from ...circuit._circuit import Circuit
+from .estimation_interface import (
+    EstimationTask,
+    EstimationTaskTransformer,
+)
+from ...hamiltonian import group_comeasureable_terms_greedy, estimate_nmeas_for_frames
+from ...utils import scale_and_discretize
+from .estimation_interface import EstimationTask
+from ...interfaces.backend import QuantumBackend, QuantumSimulator
+from ...measurement import (
+    ExpectationValues,
+    expectation_values_to_real,
+    concatenate_expectation_values,
+)
+
+greedy_grouping_with_context_selection: EstimationTaskTransformer
+uniform_shot_allocation: EstimationTaskTransformer
+proportional_shot_allocation: EstimationTaskTransformer
+
+
+def get_context_selection_circuit(
+    operator: QubitOperator,
+) -> Tuple[Circuit, IsingOperator]:
+    """Get the context selection circuit for measuring the expectation value
+    of a Pauli term.
+    TODO
```

Comment:
looks like an overlooked TODO :)

Reply:
Fixed in https://github.com/zapatacomputing/z-quantum-core/pull/259/commits/c41fe3f9eebe12e059977a0d6cffecf9ee42865b